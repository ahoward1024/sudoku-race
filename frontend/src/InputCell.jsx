import React, {Component} from 'react';
import PropTypes from 'prop-types';

class InputCell extends Component {
  constructor(props) {
    super(props);
    this.state = {
      'index': this.props.index,
      'size': this.props.size,
      'value': this.props.value
    };
    this.handleKeyPress = this.handleKeyPress.bind(this);
  }

  handleKeyPress(event) {
    if (!InputCell.isValidInput(event.key)) {
      event.preventDefault();
    }
  }

  static isValidInput(input) {
    if (input.length > 1) {

      return false;
    }
    const pattern = /^[1-9]/;

    return pattern.test(input);
  }

  render() {
    return (
      <input
        className="cell-input"
        maxLength="1"
        onKeyPress={this.handleKeyPress}
      />
    );
  }
}

InputCell.propTypes = {
  'index': PropTypes.string,
  'size': PropTypes.string,
  'value': PropTypes.string
};

export default InputCell;
