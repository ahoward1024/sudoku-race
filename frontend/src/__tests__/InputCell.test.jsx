import React from 'react';
import renderer from 'react-test-renderer';
import InputCell from '../InputCell';

test('Create an InputCell', () => {
  const inputCell = renderer.create(<InputCell/>);
  let tree = inputCell.toJSON();
  expect(tree).toMatchSnapshot();
});
