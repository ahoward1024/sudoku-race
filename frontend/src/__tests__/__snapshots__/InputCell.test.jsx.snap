// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create an InputCell 1`] = `
<input
  className="cell-input"
  maxLength="1"
  onKeyPress={undefined}
/>
`;
